### ISSUE TITLE
"dynamic" data type is recognized as "object" when using Excel COM

### ISSUE THREAD

**Issue by JohnyL** [dotnet/wpf#4203]
I have the following simple code which defines last cell with non-empty data, defines its row and gets its value:
```csharp
using Excel = Microsoft.Office.Interop.Excel;

Excel.Application xlApp = new Excel.Application { Visible = true };
Excel.Workbook xlBook = xlApp.Workbooks[1];
Excel.Worksheet sheet = xlBook.Sheets[1] as Excel.Worksheet;
int lastRow = sheet.Cells[sheet.Rows.Count, "Q"].End[Excel.XlDirection.xlUp].Row;
dynamic value = sheet.Cells[lastRow, "Q"].Value;
```
When I'm building the project using .NET 5 (5.0.101 or 5.0.200-preview.20601.7 SDK), I get the following two errors _in Output window_:
>  error CS1061: 'object' does not contain a definition for 'End' and no accessible extension method 'End' accepting a first argument of type 'object' could be found (are you missing a using directive or an assembly reference?)

> error CS1061: 'object' does not contain a definition for 'Value' and no accessible extension method 'Value' accepting a first argument of type 'object' could be found (are you missing a using directive or an assembly reference?)

Here are problems:
1. When I'm using .NET Framework 4.8, the project is built fine without any errors and code executes perfectly. In .NET 5, however, the `dynamic` data type is somehow recognized as `object`.
2. I emphasized "in Output window", because now errors do not show up in "Error List" anymore - they are now seen in "Output" window. For instance, if I remove last semicolon in my code, no errors appear in "Error List". Instead, I should build project first, and only then error appears, but in "Output" window rather than "Error List" window.

**Visual Studio Version**: 16.9.0 Preview 2.0

**Comment by JohnyL** [comment]
@jaredpar [I have already pointed](https://github.com/dotnet/wpf/issues/4203#issuecomment-778761977) out that this doesn't happen in console, but in WPF or WinForms.

**Comment by jaredpar** [comment]
@JohnyL yes but you also provided a sample project file for console apps, hence that's what I used when attempting to repro. If you knew it was WPF specific perhaps you could provide project files that demonstrate that scenario?

**Comment by ryalanms** [comment]
Thank you @JohnnyL and @jaredpar for the report and investigation.  Capitalizing the 'T' worked for me locally. 

```xml
    <ItemGroup>
             ...
            <EmbedInteropTypes>True</EmbedInteropTypes>
        </COMReference>
    </ItemGroup>
```

**Comment by JohnyL** [comment]
@jaredpar
> @JohnyL yes but you also provided a sample project file for console apps, hence that's what I used when attempting to repro.

I don't see that project file is for console application - it's for WPF.

@ryalanms With `True` (capitalized "T") - it works! But there's _more_ than that. The thing is that `true` _does_ work too. This is how I made it work:
1. When I create new application and add Excel COM reference, I get the initial entry: `<EmbedInteropTypes>true</EmbedInteropTypes>`. Building the project generates that error.
2. If I, then, change this `true` to `True`, then the project is build _successfully_.
3. And here's the trick - if I _change back_ `True` to `true`, the project is built _successfully_ too!
4. From this moment it doesn't matter whether I have `True` or `true` - the project will be build successfully.

Summary: initial changing `true` to `True` triggers something that further ignores the case of the value - whether `True` or `true`.

**Comment by jaredpar** [comment]
> I don't see that project file is for console application - it's for WPF.

Yes but unless you add in supporting files it doesn't actually trigger a WPF build and hence doesn't hit the bug.

**Comment by terrajobst** [comment]
> Summary: initial changing `true` to `True` triggers something that further ignores the case of the value - whether `True` or `true`.

Try edit a XAML after that and try to rebuild. I bet it fails again.

**Comment by JohnyL** [comment]
@terrajobst You're right! 😎 As soon as I changed XAML, that error came back. 😐 But when set to `True`, everything works fine!
